src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-list/osd-list.component.spec.ts

```diff
@@ -72,25 +110,144 @@ describe('OsdListComponent', () => {
         getTableActionComponent()
       );
       scenario = {
-        fn: () => tableActions.getCurrentButton(),
-        single: undefined,
-        empty: undefined
+        fn: () => tableActions.getCurrentButton().name,
+        single: 'Create',
+        empty: 'Create'
       };
       tableActions = permissionHelper.setPermissionsAndGetActions(1, 1, 1);
     });
 
-    it('shows no action button', () => permissionHelper.testScenarios(scenario));
+    it('shows action button', () => permissionHelper.testScenarios(scenario));
 
     it('shows all actions', () => {
-      expect(tableActions.tableActions.length).toBe(2);
+      expect(tableActions.tableActions.length).toBe(10);
       expect(tableActions.tableActions).toEqual(component.tableActions);
     });
+  });
+
+  describe('test create new OSD', () => {
+    const getPrimaryButton = () => {
+      return fixture.debugElement
+        .query(By.directive(TableActionsComponent))
+        .query(By.css('button.btn-primary')).nativeElement;
+    };
+
+    it('has a create button', () => {
+      expect(getPrimaryButton().textContent).toBe('Create');
+    });
+
+    it(
+      'navigates on click',
+      fakeAsync(() => {
+        getPrimaryButton().click();
+        tick();
+        expect(location.path()).toBe('/osd/create');
+      })
+    );
+  });
+
+  describe('test table actions in submenu', () => {
+    beforeEach(
+      fakeAsync(() => {
+        // The menu needs a click to render the dropdown!
+        const dropDownToggle = fixture.debugElement.query(By.css('.dropdown-toggle'));
+        dropDownToggle.triggerEventHandler('click', null);
+        tick();
+        fixture.detectChanges();
+      })
+    );
+
+    /**
+     * Helper function to retrieve menu item
+     * @param selector
+     */
+    const getMenuItem = (selector: string): DebugElement => {
+      return fixture.debugElement
+        .query(By.directive(TableActionsComponent))
+        .query(By.css(selector));
+    };
+
+    it('has menu entries disabled for entries without create permission', () => {
+      component.tableActions
+        .filter((tableAction) => tableAction.permission !== 'create')
+        .map((tableAction) => tableAction.name)
+        .map(TestBed.get(TableActionsComponent).toClassName)
+        .map((className) => getMenuItem(`.${className}`))
+        .forEach((debugElement) => {
+          expect(debugElement.classes.disabled).toBe(true);
+        });
```

Comment:
Thanks for your proposal but I'd prefer to keep them as they are.

Reply:
Another idea would be to use static class names in the test. This is better because it would throw an error if an action was accidentally removed - for example through a merge conflict. And it's better because it's simpler and less dynamic which is needed for stable tests.

```ts
['class1', 'class2'].forEach( (class) => expect(getMenuItem(`.${class}`).classes.disabled).toBe(true))
```

Reply:
As discussed I will create a PR after yours has been merged, that amends this change to have a smaller changeset and get opinions from all other frontend developers to get consent.